src/main/java/thrift/logic/commands/CloneCommand.java

```diff
@@ -0,0 +1,115 @@
+package thrift.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static thrift.model.transaction.TransactionDate.DATE_FORMATTER;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+
+import thrift.commons.core.Messages;
+import thrift.commons.core.index.Index;
+import thrift.logic.commands.exceptions.CommandException;
+import thrift.logic.parser.CliSyntax;
+import thrift.model.Model;
+import thrift.model.tag.Tag;
+import thrift.model.transaction.Description;
+import thrift.model.transaction.Expense;
+import thrift.model.transaction.Income;
+import thrift.model.transaction.Remark;
+import thrift.model.transaction.Transaction;
+import thrift.model.transaction.TransactionDate;
+import thrift.model.transaction.Value;
+
+/**
+ * Clones a transaction specified by its index in THRIFT.
+ */
+public class CloneCommand extends Command implements Undoable {
+
+    public static final String COMMAND_WORD = "clone";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Clones the transaction specified by its index number used in the displayed transaction list.\n"
+            + "Parameters: " + CliSyntax.PREFIX_INDEX + "INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " " + CliSyntax.PREFIX_INDEX + "1";
+
+    public static final String MESSAGE_CLONE_TRANSACTION_SUCCESS = "Cloned Transaction: %1$s";
+
+    private final Index targetIndex;
+    private Transaction clonedTransaction;
+
+    /**
+     * Creates a CloneCommand instance to clone an {@code Expense} or {@code Income
```

Comment:
There should be an empty line between description and parameter section.

Reply:
Thanks for pointing out formatting issues I overlooked!